staging/src/k8s.io/component-base/logs/registry.go

```diff
@@ -67,30 +65,38 @@ func (lfr *LogFormatRegistry) Get(name string) (logr.Logger, error) {
 }
 
 // Set specified log format logger
-func (lfr *LogFormatRegistry) Set(name string, logger logr.Logger) {
-	lfr.mu.Lock()
-	defer lfr.mu.Unlock()
+func (lfr *LogFormatRegistry) Set(name string, logger logr.Logger) error {
+	if lfr.frozen {
+		return fmt.Errorf("log format is freezed, unable to set log format")
+	}
+
 	lfr.registry[name] = logger
+	return nil
 }
 
 // Delete specified log format logger
-func (lfr *LogFormatRegistry) Delete(name string) {
-	lfr.mu.Lock()
-	defer lfr.mu.Unlock()
+func (lfr *LogFormatRegistry) Delete(name string) error {
+	if lfr.frozen {
+		return fmt.Errorf("log format is freezed, unable to delete log format")
```

Comment:
ditto

Reply:
done, sorry for delaying the PR with these small changes.